### PR TITLE
2017.7 fix for list compound targeting

### PR DESCRIPTION
**Pull request by brejoc**
### What does this PR do?

This is a follow up of #49435, but for `2017.7` instead of `develop`.

### What issues does this PR fix or reference?

Issue: #49205
PR to develop. #49435

### Previous Behavior & New Behavior

Please see the comments in #49435.

### Tests written?

No. Only very small adjustments to existing test.

### Commits signed with GPG?

Yes

### CODE REVIEW

**Comment by brejoc** [comment]
@rallytime Should I also create an PR for `2018.3`?

**Comment by rallytime** [comment]
@brejoc No, I don't think so. This will merge forward to 2018.3. Thanks!

**Review by cachedout — APPROVED** [review]

File: salt/utils/minions.py [hunk 1]
```diff
@@ -211,14 +211,15 @@ def _check_glob_minions(self, expr, greedy):  # pylint: disable=unused-argument
...
 
-    def _check_list_minions(self, expr, greedy):  # pylint: disable=unused-argument
+    def _check_list_minions(self, expr, greedy, ignore_missing=False):  # pylint: disable=unused-argument
         '''
         Return the minions found by looking via a list
         '''
         if isinstance(expr, six.string_types):
             expr = [m for m in expr.split(',') if m]
         minions = self._pki_minions()
-        return [x for x in expr if x in minions]
```
Comment (brejoc) [inline salt/utils/minions.py]: Oh, I think I mixed something up here. Maybe `2017.7` is too early and we only need that for `2018.3`. Seems like the `missing` key was not yet introduced. I'm re-checking this. Sorry!

**Comment by brejoc** [comment]
I'm closing this because of temporary mental derangement on my side. Opening up an other one for 2018.3. Sorry for the confusion!

**Pull request closed without merging**